e2e-tests/cypress/e2e/journeys/insurance/policy-and-exports/multiple-contract-policy/validation/multiple-contract-policy-validation-policy-currency-code.spec.js

```diff
@@ -0,0 +1,82 @@
+import { submitButton } from '../../../../../pages/shared';
+import { typeOfPolicyPage, multipleContractPolicyPage } from '../../../../../pages/insurance/policy-and-export';
+import partials from '../../../../../partials';
+import { ERROR_MESSAGES } from '../../../../../../../content-strings';
+import { FIELD_IDS, ROUTES } from '../../../../../../../constants';
+import getReferenceNumber from '../../../../../helpers/get-reference-number';
+import checkText from '../../../../../helpers/check-text';
+
+const { taskList } = partials.insurancePartials;
+
+const multiplePolicyFieldId = FIELD_IDS.INSURANCE.POLICY_AND_EXPORTS.POLICY_TYPE;
+const multiplePolicyField = typeOfPolicyPage[multiplePolicyFieldId].single;
+
+const { INSURANCE } = ROUTES;
+
+const {
+  INSURANCE: {
+    POLICY_AND_EXPORTS: {
+      CONTRACT_POLICY: {
+        POLICY_CURRENCY_CODE,
+      },
+    },
+  },
+} = FIELD_IDS;
+
+const {
+  INSURANCE: {
+    POLICY_AND_EXPORTS: {
+      CONTRACT_POLICY: CONTRACT_ERROR_MESSAGES,
+    },
+  },
+} = ERROR_MESSAGES;
+
+context('Insurance - Policy and exports - Multiple contract policy page - form validation - policy currency code', () => {
+  let referenceNumber;
+
+  before(() => {
+    cy.visit(INSURANCE.START, {
+      auth: {
+        username: Cypress.config('basicAuthKey'),
+        password: Cypress.config('basicAuthSecret'),
+      },
+    });
+
+    cy.submitInsuranceEligibilityAndStartApplication();
+
+    taskList.prepareApplication.tasks.policyTypeAndExports.link().click();
+
+    multiplePolicyField.input().click();
+    submitButton().click();
+
+    getReferenceNumber().then((id) => {
+      referenceNumber = id;
+      const expectedUrl = `${Cypress.config('baseUrl')}${INSURANCE.ROOT}/${referenceNumber}${INSURANCE.POLICY_AND_EXPORTS.MULTIPLE_CONTRACT_POLICY}`;
+
+      cy.url().should('eq', expectedUrl);
+    });
+  });
+
+  beforeEach(() => {
+    Cypress.Cookies.preserveOnce('_csrf');
+    Cypress.Cookies.preserveOnce('connect.sid');
+  });
+
+  const field = singleContractPolicyPage[POLICY_CURRENCY_CODE];
+
+  describe('when policy currency code is not provided', () => {
```

Comment:
just wondering since this test is so short, can this be combined with any others? would save the time of creating a new application for this test

Reply:
I like the thinking - could have a chat/session about it?

Thing is, for these kind of pages, where there are a lot of form fields and validation rules - one single field has a validation E2E test that is 150 lines long. Others like this one have a lot less.

I think if we were to combine all of these into a single "validation rules" E2E test - even if we extract the checks into seperate files/functions - I think the test would be too big.

The application creation doesn't take very long right now, but it's great to think about this and ultimately the time the entire E2E suite takes to run. When we add login, the time for E2E suite will increase. At that time I think we should have a think about anything we can do to decerease E2E time.

But I dont think this is an immediate issue we need to tackle. What do you think?

Reply:
One thing we could do is instead of going through the eligibility and "create application", maybe we could add an API call to skip that part. That'd shave off 3-5 seconds for each application test (if I correctly remember how long it takes)